packages/compass-shell/src/modules/worker-runtime.ts

```diff
@@ -52,30 +52,36 @@ export function createWorkerRuntime(
 ): typeof WorkerRuntime['prototype'] {
   const emitter = new EventEmitter();
 
-  setupLoggerAndTelemetry(
-    emitter,
-    log,
-    {
+  const loggingAndTelemetry = setupLoggingAndTelemetry({
+    bus: emitter,
+    analytics: {
       identify: () => {
         /* not needed */
       },
       // Prefix Segment events with `Shell ` to avoid event name collisions.
       // We always enable telemetry here, since the track call will
       // already check whether Compass telemetry is enabled or not.
       track: ({ event, properties }) => {
-        return track(`Shell ${event}`, properties, connectionInfo.current);
+        return track(
+          `Shell ${event as string}`,
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+          properties,
+          connectionInfo.current
+        );
       },
       flush: () => {
         return Promise.resolve(); // not needed
       },
     },
-    {
+    userTraits: {
       platform: process.platform,
       arch: process.arch,
     },
     // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-var-requires
-    require('../../package.json').version
-  );
+    mongoshVersion: require('../../package.json').version,
```

Comment:
this actually seems weird. We're passing compass shell's version as something which gets ultimately reported as `mongosh_version` in telemetry. Is that expected? @addaleax

Reply:
Yeah, it's obviously not great ... not sure what a better version would be here, though? We don't get the version reported by shell-api until we are ready to evaluate code

Reply:
Hm yeah, I guess the solution would be to not call this `mongosh_version` though it'd be too destructive to change it now and not worth it unless there's explicit need from product/analytics